frontend/src/components/table.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import {tableIcons} from "../utils/styles";
+import MaterialTable from "material-table";
+
+let tableColumns = [
+  {title: 'Id', field: 'id', editable: 'never', type: 'numeric', width: 50, cellStyle: {backgroundColor: '#fafbfc'}},
+  {title: 'Name', field: 'name'},
+  {title: 'Code', field: 'code', editable: 'never'},
+  {
+    title: 'Description',
+    field: 'description',
+    editable: 'never',
+    render: rowData =>
+      <span>{rowData.description ? rowData.description.substr(1, 25).concat('...') : null}</span>
+  },
+  {title: 'Opening Date', field: 'openingDate', editable: 'never', type: "date"},
+  {
+    title: 'Seasons',
+    field: 'seasons',
+    editable: 'never',
+    render: rowData =>
+      <span>{rowData.seasons.map(season => `${season.half} ${season.year}`).join()}</span>
+  },
+  {title: 'Special field 1', field: 'specialField1', editable: 'never'},
+  {title: 'Special field 2', field: 'specialField2', editable: 'never'}
+];
+
+const Table = ({data, onNameChange}) => {
+  return (
+    <MaterialTable
+      icons={tableIcons}
+      columns={tableColumns}
+      title="ACME® Corporation Stores Viewer"
+      options={{
+        draggable: false,
+        pageSize: 20,
+        exportButton: true,
```

Comment:
when we try to export the table as a pdf it seems to be blocking the browser and it doesn't give any feedback. do you have any idea on how could this be solved?

Reply:
Does the browser blocks indefinitely? Or after a few seconds the pdf is exported anyways?

If the latter I believe this is because the generation is happening in the main thread, and since it is a fair amount of data, it takes a fair amount of time. From reading the documentation of the I've seen that we could overwrite the CSV generation function, but not the PDF. With that said, I believe there is no easy fix for this.

Here are my sugestionsr:
1) Assessing how much do we value this feature, and if it worth keeping even with the known issue. Discarding it can be the way to go.
2) Let the user know that the freeze is normal. Either through a warning or through a loading modal (if possible)

If your browser blocks indefinitely, could you provide more information on how to replicate it? Like:
- Steps to reproduce
- Browser
- OS
That would be a good starting point